### PR TITLE
Add protective word to Medications - AB#7575

### PR DESCRIPTION
**Pull request by sslaws**
Contract.Requires apparently breaks tests - I've started to remove it.

### CODE REVIEW

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_179)

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_179)

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_179)

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_179)

File: Apps/Medication/src/Delegates/IHNClientDelegate.cs [hunk 1]
```diff
@@ -28,10 +28,11 @@ public interface IHNClientDelegate
         /// Retrieves a list of MedicationStatements that for the given phn.
         /// </summary>
         /// <param name="phn">The patient phn.</param>
+        /// <param name="protectiveWord">The clients protective word.</param>
         /// <param name="userId">The user id of the request.</param>
         /// <param name="ipAddress">The ip address of the request.</param>
         /// <returns>A List of MedicationStatement models.</returns>
-        Task<HNMessage<List<MedicationStatement>>> GetMedicationStatementsAsync(string phn, string userId, string ipAddress);
+        Task<HNMessage<List<MedicationStatement>>> GetMedicationStatementsAsync(string phn, string protectiveWord, string userId, string ipAddress);
```
Comment (FuriousLlama) [inline Apps/Medication/src/Delegates/IHNClientDelegate.cs]: kinda nit: how about using a different method for calling with a protective word
```
Task<HNMessage<List<MedicationStatement>>> GetMedicationStatementsAsync(string phn, string userId, string ipAddress);
Task<HNMessage<List<MedicationStatement>>> GetProtectedMedicationStatementsAsync(string phn, string protectiveWord, string userId, string ipAddress);
```

That way the changes would only cascade to the methods that need it, and if not call the unprotected one.

File: Apps/Medication/src/Controllers/MedicationStatementController.cs [hunk 2]
```diff
@@ -75,6 +75,23 @@ public MedicationStatementController(IAuthorizationService authorizationService,
...
+        /// Gets a json of medication record.
+        /// </summary>
+        /// <returns>The medication statement records.</returns>
+        /// <param name="hdid">The patient hdid.</param>
+        /// <param name="protectiveWord">The clients protective word for Pharmanet.</param>
+        /// <response code="200">Returns the medication statement bundle.</response>
+        /// <response code="401">The client is not authorized to retrieve the record.</response>
+        [HttpGet]
+        [Produces("application/json")]
+        [Route("{hdid}")]
```
Comment (TiagoGraf89) [inline Apps/Medication/src/Controllers/MedicationStatementController.cs]: Do we need to map the protectiveWord into the route?

Comment (sslaws) [reply Apps/Medication/src/Controllers/MedicationStatementController.cs]: We’re going to add the protective word to the header.

**Pull request merged**